scripts/fileDigest.js

```diff
@@ -0,0 +1,265 @@
+
+/**
+ * @description
+ * This CLI relies on openssl under the hood to generate and verify hashes and signatures.
+ * As such, running the script in a shell that doesn't have access to openssl CLI will fail.
+ *
+ * @example
+ * $ node ./scripts/fileDigest.js -k <path-to-private-key> [-f <path-to-file] [--verify <path-to-public-key>] [--skiprv] \
+ *      [--win] [--mac] [--linux] [--all] [--dir <path-to-dir>] [--appVersion <app-version-number>]
+ */
+
+// Configuration of command line arguments
+const { argv } = yargs
+    .option('key', {
+        alias: 'k',
+        description: 'File containing private key',
+        type: 'string',
+        demandOption: true,
+    })
+    .option('file', {
+        alias: 'f',
+        description: '(Optional) File to perform digest on.',
+        type: 'string',
+    })
+    .option('dir', {
+        description: '(Optional) Directory to find for files in.',
+        type: 'string',
+        default: path.resolve(__dirname, `../${build.directories.output}`),
+    })
+    .option('appVersion', {
+        description: '(Optional) App version to use for finding files.',
+        type: 'string',
+        default: app.version,
+    })
+    .option('verify', {
+        description: '(Optional) Verify with specified public key',
+        type: 'string',
+    })
+    .option('skiprv', {
+        description:
+            "(Optional) Skips verification using remote public key. Useful if the used private key doesn't match the current remote public key",
+        type: 'boolean',
+    })
+    .option('win', {
+        description: '(Optional) Include Windows targets as digest input',
+        type: 'boolean',
+    })
+    .option('mac', {
+        description: '(Optional) Include MacOS targets as digest input',
+        type: 'boolean',
+    })
+    .option('linux', {
+        description: '(Optional) Include Linux targets as digest input',
+        type: 'boolean',
+    })
+    .help()
+    .alias('help', 'h');
+
+const {
+    file: inputFile,
+    key: publicKeyPath,
+    verify: verifyKeyPath,
+    skiprv,
+    win,
+    mac,
+    linux,
+    dir,
+    appVersion,
+} = argv;
+
+const hashAlgorithm = 'sha256';
+
+// Produce file checksum
+async function writeChecksum(file) {
+    const { stdout } = await exec(`openssl dgst -${hashAlgorithm} ${file}`);
+
+    const hash = stdout.split('= ')[1];
+    const hashOutFile = `${file}.hash`;
+
+    fs.writeFileSync(hashOutFile, hash);
+
+    console.log('Wrote hash successfully to file:', hashOutFile);
+}
+
+/**
+ * Function to verify the newly created signature against public key
+ *
+ * @param {string} pubKeyPath public key matching private key given in argv.key (--key).
+ */
+async function verifySignature(pubKeyPath, file, sigFile) {
+    const { stdout } = await promisify(exec)(
+        `openssl dgst -${hashAlgorithm} -verify ${pubKeyPath} -signature ${sigFile} ${file}`
+    );
+
+    console.log(stdout);
+}
+
+function getPublicKey() {
```

Comment:
It would make sense to re-use the code you already have for getting the public key elsewhere, instead of having two implementations for it.

Reply:
It would, but this script is not compiled and can't use esmodules, so it gets hard to reuse code used in the app. The public key location is reused.